### ISSUE TITLE
Usage data on pools and shares not updating in the WebUI

### ISSUE THREAD

**Issue by Mazo** [rockstor/rockstor-core#1105]
http://forum.rockstor.com/t/usage-data-on-pools-and-shares-not-updated/423

As per the above thread I've been having the same issue since I installed Rockstor a few months ago. All pools and shares show 0% and 0 Bytes usage.

I've tried to take a look at what might be causing the issue and the only thing I can spot myself is that rusage and eusage are set to 0 in the JSON response for the webui. It does seem to return the correct qgroup id however it seems that maybe share_usage() in fs/btrfs.py might not be finding the correct qgroup id on line 593: if (fields[0] == share_id): and thus returning None / 0 for rusage and eusage. Pure speculation however as I haven't ran it through a debugger.

```
{id: 1, snapshots: [], pool: {id: 2,…},…}
compression_algo: "no"
eusage: 0
group: "root"
id: 1
name: "VMDK_Storage"
nfs_exports: [{id: 7, share: "VMDK_Storage", mount: "/export/VMDK_Storage", export_group: 7}]
owner: "root"
perms: "440"
pool: {id: 2,…}
pqgroup: "2015/1"
qgroup: "0/258"
replica: false
rusage: 0
size: 488386584
snapshots: []
subvol_name: "VMDK_Storage"
toc: "2016-01-16T17:34:40.481786Z"
uuid: null
```

```
[root@rockstor ~]# btrfs qgroup show /mnt2/VM_Storage
qgroupid         rfer         excl
--------         ----         ----
0/5          16.00KiB     16.00KiB
0/258        56.56GiB     56.56GiB
0/281       102.21GiB    102.21GiB
0/393        44.00KiB     44.00KiB
2015/1       56.56GiB     56.56GiB
2015/2      102.21GiB    102.21GiB
2015/3       44.00KiB     44.00KiB
```

**Comment by Mazo** [comment]
I'll be damned. It's now working fine. I believe the issue was down to a missing disk (running Rockstor as a VM, then assigning it a permanent UUID through ESXi)

After removing the missing disk the issue still persisted. After rebooting it appears to have fixed the issue. No errors were thrown in the WebUI about not being able to update the statistics.

Perhaps continue on error instead of throwing an exception so statistics for other disks work? At the very least a warning on the WebUI.

 Relevant logs below:

```
CommandException: Error running a command. cmd = ['/bin/mount', '-t', 'btrfs', '-o', 'subvol=home', u'/dev/5dcc47d29608460b9c068fef2608fa4c', '/mnt2/home']. rc = 32. stdout = ['']. stderr = ['mount: special device /dev/5dcc47d29608460b9c068fef2608fa4c does not exist', '']
[16/Jan/2016 19:16:12] ERROR [storageadmin.views.command:75] Exception while refreshing state for Pool(rockstor_rockstor). Moving on: Error running a command. cmd = ['/sbin/btrfs', 'fi', 'show', u'/dev/5dcc47d29608460b9c068fef2608fa4c']. rc = 1. stdout = ['']. stderr = ['']
[16/Jan/2016 19:16:12] ERROR [storageadmin.views.command:76] Error running a command. cmd = ['/sbin/btrfs', 'fi', 'show', u'/dev/5dcc47d29608460b9c068fef2608fa4c']. rc = 1. stdout = ['']. stderr = ['']
Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/storageadmin/views/command.py", line 67, in _refresh_pool_state
    pool_info = get_pool_info(fd.name)
  File "/opt/rockstor/src/rockstor/fs/btrfs.py", line 71, in get_pool_info
    o, e, rc = run_command(cmd)
  File "/opt/rockstor/src/rockstor/system/osi.py", line 89, in run_command
    raise CommandException(cmd, out, err, rc)
CommandException: Error running a command. cmd = ['/sbin/btrfs', 'fi', 'show', u'/dev/5dcc47d29608460b9c068fef2608fa4c']. rc = 1. stdout = ['']. stderr = ['']
[16/Jan/2016 19:16:12] ERROR [storageadmin.middleware:35] Exception occured while processing a request. Path: /api/commands/bootstrap method: POST
[16/Jan/2016 19:16:12] ERROR [storageadmin.middleware:36] Error running a command. cmd = ['/bin/mount', '-t', 'btrfs', '-o', 'subvol=home', u'/dev/5dcc47d29608460b9c068fef2608fa4c', '/mnt2/home']. rc = 32. stdout = ['']. stderr = ['mount: special device /dev/5dcc47d29608460b9c068fef2608fa4c does not exist', '']
Traceback (most recent call last):
  File "/opt/rockstor/eggs/Django-1.6.11-py2.7.egg/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/rockstor/eggs/Django-1.6.11-py2.7.egg/django/views/decorators/csrf.py", line 57, in wrapped_view
    return view_func(*args, **kwargs)
  File "/opt/rockstor/eggs/Django-1.6.11-py2.7.egg/django/views/generic/base.py", line 69, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/rockstor/eggs/djangorestframework-3.1.1-py2.7.egg/rest_framework/views.py", line 452, in dispatch
    response = self.handle_exception(exc)
  File "/opt/rockstor/eggs/djangorestframework-3.1.1-py2.7.egg/rest_framework/views.py", line 449, in dispatch
    response = handler(request, *args, **kwargs)
  File "/opt/rockstor/eggs/Django-1.6.11-py2.7.egg/django/db/transaction.py", line 371, in inner
    return func(*args, **kwargs)
  File "/opt/rockstor/src/rockstor/storageadmin/views/command.py", line 84, in post
    import_shares(p, request)
  File "/opt/rockstor/src/rockstor/storageadmin/views/share_helpers.py", line 127, in import_shares
    mount_share(nso, '%s%s' % (settings.MNT_PT, s))
  File "/opt/rockstor/src/rockstor/fs/btrfs.py", line 210, in mount_share
    return run_command(mnt_cmd)
  File "/opt/rockstor/src/rockstor/system/osi.py", line 89, in run_command
    raise CommandException(cmd, out, err, rc)
CommandException: Error running a command. cmd = ['/bin/mount', '-t', 'btrfs', '-o', 'subvol=home', u'/dev/5dcc47d29608460b9c068fef2608fa4c', '/mnt2/home']. rc = 32. stdout = ['']. stderr = ['mount: special device /dev/5dcc47d29608460b9c068fef2608fa4c does not exist', '']
```

**Comment by schakrava** [comment]
Really nice investigative work @Mazo! I wish you best of luck and like to welcome you to open pull requests.

But you are right that your initial speculation is incorrect. Two things were going on. share usage was only being updated when you first visit the UI and give it a few seconds to propagate to the UI -- not ideal. Thought reboot is only a broad correlation, it's really due to another trigger in data_collector. The second thing is that Pool usage was not at all getting updated because there was an older suboptimal implimentation which was removed a while back. Both of these issues are now fixed in #1133 

The mount problem in the traceback in your last comment is unrelated.
